### PR TITLE
Add next page button

### LINKED ISSUE #2: Searching for pictures only displays the first page of Pinterest results (22 ± 3 posts)
## Current output

Only around 20-25 pictures are shown when searching for any query. This happens on both instances (ahwx and revvy). Using the original site, the pictures appear gradually (~20 load, then if the user scrolls, more posts appear).

## Steps to reproduce

1. Use any instance (for example pinternet.ahwx.org),
2. Search for any query with moderate interest.

## Expected output

Similar behavior to the original site.

### PR DESCRIPTION
**Pull request by qminlh**
Added a "Next page" button.

## Technical details

First of all, in the description below I'm mentioning "search data" - it's a JSON object that can be defined as follows (TypeScript-style definition):

```javascript
{
	options: {
		query: string,  // search query
		bookmarks?: string[],  // undefined for first page; if defined, it's an array containing a single string that is a reference to current position
		// irrelevant stuff
	},
	// irrelevant stuff
}
```

Requests for new images are made to the URL "https://www.pinterest.com/resource/BaseSearchResource/get/". First request is done as a GET method and contains URL-encoded search data as a query parameter with key `data` (that part is already implemented), while further requests are done with POST method and the query parameter is moved to request body together with the key, with encoding unchanged.

In POST requests, search data also needs to contain `bookmarks` - an array containing a single string, which serves as a reference to user's current position/page. The string is located in previous response, in `.resource_response.bookmark`. These requests also need a `csrftoken` cookie, which is received on a first request.

## Implementation notes

* I store CSRF token in the URL and not as a cookie, so that by design a new token is generated with each search - this prevents fingerprinting.
* Response object contains the entire `bookmarks` array in `.resource.bookmarks` - it's mostly the same as the string that I used, with a different behavior only on last page (`.resource_response.bookmarks` becomes undefined, while `.resource.bookmarks` changes value to `["-end-"]`). Choosing `.resource_response.bookmark` was an arbitrary decision.

## Additional notes

* I also added "www." to the beginning of the URL to avoid unnecessary extra request, as the pinterest.com URL redirects to www.pinterest.com.
* Full-fledged pagination may or may not be possible - research on the structure of `bookmark` is needed.
* I don't have much PHP experience, so I would strongly recommend a security-oriented check.

Fixes #2

### CODE REVIEW

File: search.php [hunk 1]
```diff
@@ -45,11 +121,21 @@
...
+    return $result;
 };
 
-$images = $search("$query");
+$result = $search($query, $bookmark);
+
+if ($result->bookmark != null)
+{
+    $bookmark_encoded = urlencode($result->bookmark);
+    echo "<h2 style=\"text-align: center;\"><a href=\"/search.php?q=$query&bookmark=$bookmark_encoded&csrftoken=$csrftoken\">Next page</a></h2><br><br><br>";
```
Comment (ouuan) [inline search.php:137]: It seems that this is XSS vulnerable 🤔

Comment (ouuan) [reply search.php:137]: I tried searching `"><script>alert(1)</script>` and it succeeded.

Comment (qminlh) [reply search.php]: Fixed, thank you!

**Comment by Ahwxorg** [comment]
Is the XSS vuln patched in the latest commit? @qminlh

**Comment by qminlh** [comment]
> Is the XSS vuln patched in the latest commit? @qminlh

Yes

**Comment by Ahwxorg** [comment]
Alright, merging it :)

**Pull request merged**